docs/sources/write/shortcodes/index.md

```diff
@@ -437,18 +509,28 @@ Set the reference at the end of the page as follows:
 
```

Comment:
@jdbaldry I'm finding some patterns that aren't explicitly included in the docs here or in what's already published... and maybe it's worth mentioning or clarifying?

Assume a file at: `/docs/sources/agent/static/example.md`, then:

```
{{% docs/reference %}}
[example]:  "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/example"
{{% docs/reference %}}
```

will end in a 404 error when the docs are built.... 

but this works:

```
{{% docs/reference %}}
[example]:  "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/example.md"
{{% docs/reference %}}
```

But this doesn't carry over to `_index.md` files

```
{{% docs/reference %}}
[link]:  "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/_index.md"
{{% docs/reference %}}
```

will resolve fine as will

```
{{% docs/reference %}}
[link]:  "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow"
{{% docs/reference %}}
```

Reply:
The `docs/reference` shortcode uses the Hugo `relref` function under the hood and my understanding is that both of your examples should work the same.

For example, linking to `errata.md`, in https://github.com/grafana/writers-toolkit/blob/jdb%2F2023-10-update-linking-guidance/docs/sources/review/doc-validator/_index.md?plain=1#L28